### PR TITLE
add list functions

### PR DESCRIPTION
**Pull request by jievince**

### CODE REVIEW

File: src/common/function/FunctionManager.cpp [hunk 1]
```diff
@@ -1155,6 +1168,118 @@ FunctionManager::FunctionManager() {
             return Value::kNullValue;
         };
     }
+    {
+        auto &attr = functions_["keys"];
+        attr.minArity_ = 1;
+        attr.maxArity_ = 1;
+        attr.isPure_ = true;
+        attr.body_ = [](const auto &args) -> Value {
+            std::set<std::string> tmp;
```
[the remark this answers is not in the document]
Comment (jievince) [reply src/common/function/FunctionManager.cpp:1174]: keys() return ordered result

**Review by Aiee — DISMISSED** [review]

File: src/common/function/FunctionManager.cpp [hunk 2]
```diff
@@ -1317,6 +1333,67 @@ FunctionManager::FunctionManager() {
...
+        auto &attr = functions_["tail"];
+        attr.minArity_ = 1;
+        attr.maxArity_ = 1;
+        attr.isPure_ = true;
+        attr.body_ = [](const auto &args) -> Value {
+            if (!args[0].isList()) {
+                return Value::kNullBadType;
+            }
+            auto& list = args[0].getList();
+            List result(std::vector<Value>(list.values.begin()+1, list.values.end()));
```
Comment (Shylock-Hg) [inline src/common/function/FunctionManager.cpp:1393]: Check is empty first.

Comment (jievince) [reply src/common/function/FunctionManager.cpp:1396]: Fixed, thanks

File: src/common/function/FunctionManager.cpp [hunk 3]
```diff
@@ -189,6 +190,15 @@ std::unordered_map<std::string, std::vector<TypeSignature>> FunctionManager::typ
                    TypeSignature({Value::Type::PATH}, Value::Type::VERTEX), }},
     {"endNode", {TypeSignature({Value::Type::EDGE}, Value::Type::VERTEX),
                  TypeSignature({Value::Type::PATH}, Value::Type::VERTEX), }},
+    {"keys", {TypeSignature({Value::Type::VERTEX}, Value::Type::LIST),
+              TypeSignature({Value::Type::EDGE}, Value::Type::LIST),
+              TypeSignature({Value::Type::MAP}, Value::Type::LIST), }},
+    {"nodes", {TypeSignature({Value::Type::PATH}, Value::Type::LIST), }},
+    // reduce needs List comprehension expression which is not ready now
```
Comment (Shylock-Hg) [inline src/common/function/FunctionManager.cpp:197]: Add TODO or delete the comment.

**Review by Shylock-Hg — APPROVED** [review]

**Review by Aiee — APPROVED** [review]

**Review by czpmango — APPROVED** [review]

**Pull request merged**